.github/tests-visual/visual-record.js

```diff
@@ -107,50 +110,76 @@ async function main() {
-        'mediapool_sync.png': START_URL + '?page=mediapool/sync',
-        'templates.png': START_URL + '?page=templates',
-        'templates_edit.png': START_URL + '?page=templates&function=edit&template_id=1',
-        'modules_modules.png': START_URL + '?page=modules/modules',
-        'modules_actions.png': START_URL + '?page=modules/actions',
-        'users_users.png': START_URL + '?page=users/users',
-        'users_edit.png': START_URL + '?page=users/users&user_id=1',
-        'users_roles.png': START_URL + '?page=users/roles',
-        'users_role_add.png': START_URL + '?page=users/roles&func=add&default_value=1',
-        'packages.png': START_URL + '?page=packages',
-        'system_settings.png': START_URL + '?page=system/settings',
-        'system_log.png': START_URL + '?page=system/log/redaxo',
-        'system_report.png': START_URL + '?page=system/report/html',
-        'backup_export.png': START_URL + '?page=backup/export',
-        'backup_import.png': START_URL + '?page=backup/import',
-        'cronjob_cronjobs.png': START_URL + '?page=cronjob/cronjobs',
-        'media_manager_types.png': START_URL + '?page=media_manager/types',
-        'media_manager_types_edit.png': START_URL + '?page=media_manager/types&type_id=1&effects=1',
-        'media_manager_settings.png': START_URL + '?page=media_manager/settings',
-        'metainfo_articles.png': START_URL + '?page=metainfo/articles',
-        'metainfo_categories.png': START_URL + '?page=metainfo/categories',
-        'metainfo_media.png': START_URL + '?page=metainfo/media',
-        'metainfo_clangs.png': START_URL + '?page=metainfo/clangs',
-        'phpmailer_config.png': START_URL + '?page=phpmailer/config',
-    };
-
-    for (var fileName in config) {
-        await page.goto(config[fileName]);
-        await new Promise(res => setTimeout(() => res(), 1000));
-        await createScreenshot(page, fileName);
+    if (isSetup) {
+
+        // setup step 1
+        await page.goto(START_URL, { waitUntil: 'load' });
+        await createScreenshot(page, 'setup.png');
+
+        // setup steps 2-7
+        for (var step = 2; step <= 7; step++) {
+            await page.goto(START_URL + '?page=setup&lang=de_de&step=' + step, { waitUntil: 'load' });
+            await createScreenshot(page, 'setup_' + step + '.png');
+        }
+
+    } else {
+
+        // login page
+        await page.goto(START_URL, { waitUntil: 'load' });
+        await page.waitForTimeout(1000); // CSS animation
+        await createScreenshot(page, 'login.png');
+
+        // login successful
+        await page.type('#rex-id-login-user', 'myusername');
+        await page.type('#rex-id-login-password', '91dfd9ddb4198affc5c194cd8ce6d338fde470e2'); // sha1('mypassword')
+        await page.$eval('#rex-form-login', form => form.submit());
+        await page.waitForTimeout(1000);
+        await createScreenshot(page, 'index.png');
+
+        // all pages
+        var config = {
```

Comment:
der übersichtlichkeit halber könnte man dieses array in eine separate methode verlagern und dort nur returnen, sodass die methode hier lesbarer wird

Reply:
(oder zu den anderen config parametern ganz an den anfang des scripts - statt neue methode/function)

Reply:
Erledigt. :)